Fix: OpenAI , azure, gemini tests and update credentials

priyansh4320 opened the pull request:
<!-- Thank you for your contribution! Please review https://docs.ag2.ai/latest/docs/contributor-guide/contributing before opening a pull request. -->

<!-- Please add a reviewer to the assignee section when you create a PR. If you don't have the access to it, we will shortly find a reviewer and assign them to your PR. -->

## Why are these changes needed?
This PR fixes deployments failure tests for core-llm-tests, includes:
- openai tests
- azure tests
- gemini realtime tests
- openai realtime tests
- anthropic tests
<!-- Please give a short summary of the change and the problem this solves. -->

## Related issue number
#2196 
<!-- For example: "Closes #1234" -->

## Checks

- [x] I've included any doc changes needed for https://docs.ag2.ai/. See https://docs.ag2.ai/latest/docs/contributor-guide/documentation/ to build and test documentation locally.
- [x] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [x] I've made sure all auto checks have passed.

marklysze committed:
Update sonnet model to 4.5

priyansh4320 committed:
fix: openai, gemini realtime

priyansh4320 committed:
update: credentials

priyansh4320 committed:
fix: realtime agent tool registry

priyansh4320 committed:
fix: assistant agent test

priyansh4320 committed:
Merge branch 'main' into fix-openai-tests

Commit:
[skip ci] docs: fix outdated docs

priyansh4320 committed:
fix: credentials

priyansh4320 committed:
fix: azure tests

Commit:
[skip ci] docs: fix outdated docs

priyansh4320 committed:
Merge branch 'fix-openai-tests' into docs/jgr-autofix-ag_2-2188

priyansh4320 committed:
fix: pre-commit

priyansh4320 committed:
Merge pull request #2194 from ag2ai/docs/jgr-autofix-ag_2-2188

docs: for pull #2188, fixed 1 outdated doc(s)

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/ag2ai/ag2?pullRequest=2188) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/ag2ai/ag2?pullRequest=2188) before we can accept your contribution.<br/>**2** out of **3** committers have signed the CLA.<br/><br/>:white_check_mark: marklysze<br/>:white_check_mark: priyansh4320<br/>:x: joggrbot[bot]<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/ag2ai/ag2?pullRequest=2188) it.</sub>

randombet commented on autogen/oai/oai_models/chat_completion_message.py:
```diff
@@ -55,7 +57,9 @@ class FunctionCall(BaseModel):
 
 class ChatCompletionMessage(BaseModel):
-    content: str | dict[str, Any] | list[dict[str, Any]] | None = None
+    content: str | dict[str, Any] | list[dict[str, Any]] | None = Field(
```
Why change this?

priyansh4320 replied:
a test broke due to conflict in str type for oai sdk &  and list[dict[str,Any]] .
so the current fix resolves this issue. so that is why I have added a field to ChatCompletionMessage

randombet replied:
Which test?

randombet commented on website/docs/contributor-guide/setup-development-environment.mdx:
```diff
@@ -6,9 +6,29 @@ title: Setting up Development Environment
-=== "Using OAI_CONFIG_LIST"
+=== "Using environment variables (recommended/main method)"
 
-    AG2 uses an environment variable called `OAI_CONFIG_LIST` in JSON format to store the LLM keys. `OAI_CONFIG_LIST` is a list of dictionaries where each dictionary contains the following keys:
```
json config file is not only for keys but for other LLM params, e.g. temperature.
Is this PR only change for the key fetching logic?

priyansh4320 replied:
```
As an alternative, AG2 also supports an environment variable called OAI_CONFIG_LIST in JSON format to store the LLM keys. OAI_CONFIG_LIST is a list of dictionaries where each dictionary contains the following keys:
```
the OAI_CONFIG_LIST.json is mention in the mdx file, joggr bot added this pull since i have updated credetials to Secrets as priority with fallback on OAI_CONFIG_LIST

 the OAI_CONFIG_LIST support for LLM params is present in the documentation while the documentation in my PR is only for development setup, ie., website/docs/contributor-guide/setup-development-environment.mdx
which only needs information about LLM  keys not params

priyansh4320 commented on website/docs/user-guide/advanced-concepts/llm-configuration-deep-dive.mdx:
```diff
@@ -6,7 +6,22 @@ In this deep-dive we run through the LLM configuration in depth, including the u
 
 # LLM Configuration
 
-In AG2, agents use LLMs as key components to understand and react. To configure an agent's access to LLMs, you can specify an `llm_config` argument in its constructor. For example, the following snippet shows a configuration that uses `gpt-4.1`:
```
ok

priyansh4320 committed:
fix: dev setup mdx

Revert "Merge pull request #2194 from ag2ai/docs/jgr-autofix-ag_2-2188"

This reverts commit 3aa474c7be955f9b58b8b3b7d8b19287992129d5, reversing
changes made to 4d9de80ac2da03dc7912d312e1d81594406c63e1.

fix: user-guide

randombet reviewed (approved):
LGTM

marklysze reviewed (approved):
Great job @priyansh4320!
